altinn-client/src/main/java/no/difi/meldingsutveksling/AltinnWsClient.java

```diff
@@ -54,29 +61,47 @@ public void send(UploadRequest request) {
 
     private void upload(UploadRequest request, String senderReference) {
 
-        BrokerServiceExternalBasicStreamedSF brokerServiceExternalBasicStreamedSF
-                = new BrokerServiceExternalBasicStreamedSF(configuration.getStreamingServiceUrl());
-
-        IBrokerServiceExternalBasicStreamed streamingService = brokerServiceExternalBasicStreamedSF.getBasicHttpBindingIBrokerServiceExternalBasicStreamed();
-
-        BindingProvider bp = (BindingProvider) streamingService;
-        bp.getRequestContext().put(BindingProvider.ENDPOINT_ADDRESS_PROPERTY, configuration.getStreamingServiceUrl().toString());
-        bp.getRequestContext().put(JAXWSProperties.REQUEST_TIMEOUT, REQUEST_TIMEOUT);
-        bp.getRequestContext().put(JAXWSProperties.CONNECT_TIMEOUT, CONNECT_TIMEOUT);
-
         try {
             StreamedPayloadBasicBE parameters = new StreamedPayloadBasicBE();
+            parameters.setDataStream(getDataHandler(request));
+
+            CompletableFuture<Void> altinnUpload = CompletableFuture.runAsync(
+                    () -> uploadToAltinn(request, senderReference, parameters),
+                    taskExecutor);
+
+            try {
+                log.debug("Blocking main thread to wait for upload..");
+                altinnUpload.get();
+            } catch (InterruptedException | ExecutionException e) {
+                throw new MeldingsUtvekslingRuntimeException("Error waiting for upload thread to finish", e);
+            }
+        } catch (Exception e) {
+            Audit.error("Message failed to upload to altinn", request.getMarkers(), e);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "Message failed to upload to altinn" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1192)